File: app/view_models/ranking_report.rb
# frozen_string_literal: true

class RankingReport
  attr_reader \
    :report_date,
    :top_ytd,
    :top_monthly,
    :top_weekly,
    :top_daily,
    :lowest_risk,
    :highest_risk,
    :largest_drawdown

  def initialize(
    report_date:,
    top_ytd:,
    top_monthly:,
    top_weekly:,
    top_daily:,
    lowest_risk:,
    highest_risk:,
    largest_drawdown:
  )
    @report_date = report_date
    @top_ytd = top_ytd
    @top_monthly = top_monthly
    @top_weekly = top_weekly
    @top_daily = top_daily
    @lowest_risk = lowest_risk
    @highest_risk = highest_risk
    @largest_drawdown = largest_drawdown

    freeze
  end
end

File: mobile/src/models/RankingReport.ts
import type { FundRanking } from './FundRanking';

export interface RankingReport {
  report_date: string;

  top_ytd: FundRanking[];

  top_monthly: FundRanking[];

  top_weekly: FundRanking[];

  top_daily: FundRanking[];

  lowest_risk: FundRanking[];

  highest_risk: FundRanking[];

  worst_drawdown: FundRanking[];
}
